code/modules/paperwork/clipboard.dm

```diff
@@ -1,154 +1,157 @@
+#define PAPERWORK	1
+#define PHOTO		2
+#define IS_PEN(x)	(istype(x, /obj/item/weapon/pen))
+
 /obj/item/weapon/clipboard
 	name = "clipboard"
+	desc = "It looks like you're writing a letter. Want some help?"
 	icon = 'icons/obj/bureaucracy.dmi'
 	icon_state = "clipboard"
 	item_state = "clipboard"
-	throwforce = 0
 	w_class = WEIGHT_CLASS_SMALL
-	throw_speed = 3
-	throw_range = 10
-	var/obj/item/weapon/pen/haspen		//The stored pen.
-	var/obj/item/weapon/toppaper	//The topmost piece of paper.
+	var/obj/item/weapon/pen/containedpen
+	var/obj/item/weapon/toppaper
 	slot_flags = SLOT_BELT
 	burn_state = FLAMMABLE
 
 /obj/item/weapon/clipboard/New()
+	..()
 	update_icon()
 
-/obj/item/weapon/clipboard/MouseDrop(obj/over_object as obj) //Quick clipboard fix. -Agouri
-	if(ishuman(usr))
-		var/mob/M = usr
-		if(!(istype(over_object, /obj/screen) ))
-			return ..()
-
-		if(!M.restrained() && !M.stat)
-			switch(over_object.name)
-				if("r_hand")
-					M.unEquip(src)
-					M.put_in_r_hand(src)
-				if("l_hand")
-					M.unEquip(src)
-					M.put_in_l_hand(src)
-
-			add_fingerprint(usr)
+/obj/item/weapon/clipboard/verb/removePen()
+	set category = "Object"
+	set name = "Remove clipboard pen"
+	if(!ishuman(usr) || usr.incapacitated())
+		return
+	penPlacement(FALSE, containedpen)
+
+/obj/item/weapon/clipboard/proc/isPaperwork(obj/item/weapon/W) //This could probably do with being somewhere else but for now it's fine here.
+	if(istype(W, /obj/item/weapon/paper) || istype(W, /obj/item/weapon/paper_bundle))
+		return PAPERWORK
+	if(istype(W, /obj/item/weapon/photo))
+		return PHOTO
+
+/obj/item/weapon/clipboard/proc/checkTopPaper()
+	if(toppaper.loc != src) //Oh no! We're missing a top sheet! Better get another one to be at the top.
+		toppaper = locate(/obj/item/weapon/paper) in src
+		if(!toppaper) //In case there's no paper, try find a paper bundle instead (why is paper_bundle not a subtype of paper?)
+			toppaper = locate(/obj/item/weapon/paper_bundle) in src
+
+/obj/item/weapon/clipboard/examine(mob/user)
+	if(..(user, 1) && toppaper)
+		toppaper.examine(user)
+
+obj/item/weapon/clipboard/proc/penPlacement(placing, obj/item/weapon/pen/P)
```

Comment:
This should really have a user argument, since you actually do call it both where you have a predefined `user` argument and where you just have a `usr` verb.